omod/src/main/webapp/resources/scripts/navigator/navigatorHandlers.js

```diff
@@ -307,18 +314,20 @@ var clickedFieldHandler = function(fields, field, event) {
     var clickedFieldIndex = _.indexOf(fields, field);
     var shouldSelectClickedField = true;
     if(clickedFieldIndex > currentFieldIndex) {
-        for(var i=currentFieldIndex; i<clickedFieldIndex; i++) {
+        for(var i=currentFieldIndex == -1 ? 0 : currentFieldIndex; i<clickedFieldIndex; i++) {
             shouldSelectClickedField = fields[i].isValid() && shouldSelectClickedField;
         }
     }
 
     // call exit handler if validation has passed
-    shouldSelectClickedField = shouldSelectClickedField && currentField.onExit();
+    if (currentField) {
+      shouldSelectClickedField = shouldSelectClickedField && currentField.onExit();
+    }
 
     if(!shouldSelectClickedField) {
-        currentField.select();
+        currentField && currentField.select();
     } else {
-        currentField.toggleSelection();
+        currentField && currentField.toggleSelection();
         field.toggleSelection();
     }
```

Comment:
Instead of introducing the `currentField &&`s, could you just move all this stuff into the `if (currentField)` block? Or is it possible that `shouldSelectClickedField && !currentField`?

Reply:
The latter may be possible, I'm not 100% sure.  This change was due to a bug I encountered while testing.  I think the code was written with the assumption that the currentField could never be null, and that assumption is wrong.  I started doing a bigger refactor but then held off because this wasn't what I was actually working on and didn't want to inadvertently break something.